wasm/src/thread_pool/mod.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the Aleo SDK library.
+
+// The Aleo SDK library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Aleo SDK library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Aleo SDK library. If not, see <https://www.gnu.org/licenses/>.
+
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+use rayon::ThreadBuilder;
+use futures::future::try_join_all;
+use std::future::Future;
+use spmc::{channel, Sender, Receiver};
+
+
+#[wasm_bindgen(inline_js = r###"
+    export function spawnWorker(url, module, memory, address) {
+        return new Promise((resolve) => {
+            const worker = new Worker(url, {
+                type: "module",
+            });
+
+            worker.addEventListener("message", (event) => {
+                resolve(worker);
+            }, {
+                capture: true,
+                once: true,
+            });
+
+            worker.postMessage({
+                module,
+                memory,
+                address,
+            });
+        });
+    }
+"###)]
+extern "C" {
+    #[wasm_bindgen(js_name = spawnWorker)]
+    fn spawn_worker(
+        url: &web_sys::Url,
+        module: &JsValue,
+        memory: &JsValue,
+        address: *const Receiver<ThreadBuilder>,
```

Comment:
This is the address of the pointer right?

Reply:
Yes, it's a pointer to the Rayon thread. We can't send Rust objects to another Worker, but we can send a pointer (integer) to another Worker. So it sends the pointer to the Worker, so that way the Worker can dereference the pointer and access the Rayon thread.